From 3a47113d98acf1ffcc1925f803cf318784a5dff2 Mon Sep 17 00:00:00 2001
From: vamsikrishnak <vamsikrishnak@ami.com>
Date: Sun, 10 Dec 2023 04:29:09 -0500
Subject: [PATCH] Fixed LocalRole Patch error in redfish

Root Cause: This issue is occur due upstream
code changes in accountService file .

This commmit fixes the error occur during
patch call of localRole property

Tested:
    The changes were tested and working as expected

Signed-off-by: vamsikrishnak <vamsikrishnak@ami.com>
---
 redfish-core/lib/account_service.hpp | 2 +-
 1 file changed, 1 insertion(+), 1 deletion(-)

diff --git a/redfish-core/lib/account_service.hpp b/redfish-core/lib/account_service.hpp
index ec10c915..d7f74ec6 100644
--- a/redfish-core/lib/account_service.hpp
+++ b/redfish-core/lib/account_service.hpp
@@ -463,7 +463,7 @@ inline void handleRoleMapPatch(
                         *crow::connections::systemBus, ldapDbusService,
                         roleMapObjData[index].first,
                         "xyz.openbmc_project.User.PrivilegeMapperEntry",
-                        "Privilege", *localRole,
+                        "Privilege", getPrivilegeFromRoleId(*localRole),
                         [asyncResp, roleMapObjData, serverType, index,
                          localRole](const boost::system::error_code& ec,
                                     const sdbusplus::message_t& msg) {
-- 
2.34.1

